cms/updates/0.0.5-evidence-types-update.js

```diff
@@ -0,0 +1,120 @@
+const keystone = require("keystone"),
+	async = require("async"),
+	log4js = require("log4js"),
+	fs = require("fs"),
+	path = require("path");
+
+const logger = log4js.getLogger();
+const evidenceTypeList = keystone.list("EvidenceType");
+
+const deleteEvidenceType = async (evidenceType, done) => {
+	let eTitle = evidenceType.title;
+	logger.info(`Deleting evidence type: ${eTitle}`);
+
+	try {
+		await evidenceTypeList.model.findOne({ title: { "$regex": eTitle, "$options": "i" } }).remove();
+		done();
+	} catch (err) {
+		logger.error(`Error deleting '${eTitle}'`, err);
+		done(err);
+	}
+
+	logger.info(`Successfully deleted '${eTitle}'`);
+};
+
+const updateEvidenceType = async (jsonldEvidenceType, done) => {
+	let masEvidenceType = await evidenceTypeList.model.findOne({key: jsonldEvidenceType["@id"] }).exec();
+	if(!masEvidenceType)
+		return;
+
+	let newTitleShort = jsonldEvidenceType["prefLabel"]["@value"];
+	let newTitle = masEvidenceType.broaderTitle + " - " + newTitleShort;
+	let oldTitle = masEvidenceType.title;
+
+	if(newTitle != oldTitle && !oldTitle.includes(newTitleShort)){
+
+		logger.info(`Updating evidence type '${oldTitle}' to '${newTitle}'`);
+
+		try {
+			var docToUpdate = await evidenceTypeList.model.findOne({ key: jsonldEvidenceType["@id"] });
+			docToUpdate.title = newTitle;
+			docToUpdate.key = jsonldEvidenceType["@id"];
```

Comment:
why are you setting the key here?

Reply:
We are resetting the key here (not the primary key btw thats the _id) because the keys value is somewhat representative of the title.

Reply:
If we're updating an item here, then by definition the key is the same, so it does seem odd to be setting it.

Reply:
If a title is updated then the key is also updated. For examplpe it would not make sense to have a item titled newtitle with the key of oldtitle.

Reply:
If a title is updated then the key is also updated. For examplpe it would not make sense to have a item titled newtitle with the key of oldtitle. It's interesting that the key is not used for querying the data, thats the _id.